src/php/Emitter/OutputEmitter/NodeEmitter/ApplyEmitter.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phel\Emitter\OutputEmitter\NodeEmitter;
+
+use Phel\Ast\ApplyNode;
+use Phel\Ast\Node;
+use Phel\Ast\PhpVarNode;
+use Phel\Emitter\OutputEmitter\NodeEmitter;
+
+final class ApplyEmitter implements NodeEmitter
+{
+    use WithOutputEmitter;
+
+    public function emit(Node $node): void
+    {
+        assert($node instanceof ApplyNode);
```

Comment:
Ideally, this line shouldn't be necessary at this point.
Ideally, we should be able to define the node-child-type in the method signature, like
```php
final class ApplyEmitter implements NodeEmitter
{
    use WithOutputEmitter;

    public function emit(ApplyNode $node): void
    { // ...
```
This is not currently possible with PHP 7.4. But I think it will be allowed with PHP 8.

Reply:
Yes, I think this will possible on PHP 8.
What about creating an `issue` in order to remember the change when we upgrade the PHP version.

By the way, do you have in mind update to version 8 when it is available or keep on on version 7.4? @jenshaase

Reply:
I think it makes total sense to update to version PHP 8. Using docker it should be something trivial :)

Reply:
I would like to support 7.4 a little bit longer. One of my goals was that Phel can be executed on cheap hosting providers and they normally need a little bit of time to support the newest version of PHP.